homeassistant/components/izone/discovery.py

```diff
@@ -0,0 +1,133 @@
+"""Internal discovery service for  iZone AC."""
+
+import logging
+from asyncio import Event
+from typing import Dict
+
+import pizone
+
+from homeassistant.const import CONF_EXCLUDE, EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .climate import ControllerDevice
+from .constants import DATA_ADD_ENTRIES, DATA_CONFIG, DATA_DISCOVERY_SERVICE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_start_discovery_service(hass: HomeAssistantType):
+    """Set up the pizone internal discovery."""
+    class DiscoveryService(pizone.Listener):
```

Comment:
Why construct this class in the function instead of the module scope? The pizone import is already at the module level. And hass can be passed in to init.

Reply:
Probably hanging from when imports had to be inside functions rather than at module scope. Will fix it.